ogr/ogrfeature.cpp

```diff
@@ -456,11 +456,12 @@ OGRFeatureDefnH OGR_F_GetDefnRef(OGRFeatureH hFeat)
 OGRErr OGRFeature::SetGeometryDirectly(OGRGeometry *poGeomIn)
 
 {
-    if (GetGeomFieldCount() > 0)
-        return SetGeomFieldDirectly(0, poGeomIn);
+    if (poGeomIn == GetGeometryRef())
+    {
+        return OGRERR_NONE;
+    }
 
-    delete poGeomIn;
-    return OGRERR_FAILURE;
+    return SetGeometry(std::unique_ptr<OGRGeometry>(poGeomIn));
```

Comment:
micro-optimization to avoid an extra function call:
```suggestion
    return SetGeomField(0, std::unique_ptr<OGRGeometry>(poGeomIn));
```

Reply:
done